### PR TITLE
Add Activision dissector.

### PR DESCRIPTION
**Pull request by utoni**
Signed-off-by: lns <matzeton@googlemail.com>

### CODE REVIEW

File: src/lib/protocols/activision.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+void ndpi_search_activision(struct ndpi_detection_module_struct *ndpi_struct,
+                            struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search activision\n");
+
+  if (packet->payload_packet_len < 18)
```
Comment (IvanNardi) [inline src/lib/protocols/activision.c:45]: You check 18...

File: src/lib/protocols/activision.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          return;
+        }
+        break;
+      default:
+        NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+        return;
+    }
+
+    if (ntohs(get_u_int16_t(packet->payload, 17)) == 0xc0a8 &&
+        ntohl(get_u_int32_t(packet->payload, 19)) == 0x0015020c)
```
Comment (IvanNardi) [inline src/lib/protocols/activision.c:75]: but you read past after 19 here

File: src/lib/protocols/activision.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  if (packet->payload_packet_len < 18)
+  {
+    NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    return;
+  }
+
+  if (flow->packet_direction_counter[packet->packet_direction] == 1)
+  {
+    switch (packet->packet_direction)
```
Comment (IvanNardi) [inline src/lib/protocols/activision.c]: Not sure if this `switch` is necessary. If `packet->packet_direction` is not 0 or 1, we have some *seriuos* issues  everywhere

File: src/lib/protocols/activision.c (hunk 1 above)
Comment (utoni) [reply src/lib/protocols/activision.c:45]: whoops! thanks

**Review by IvanNardi — APPROVED** [review]

**Pull request merged**